crates/kernel/src/tasks/scheduler.rs

```diff
@@ -140,16 +141,16 @@ struct TaskControl {
     suspended: bool,
     waiting_input: Option<Uuid>,
     resume_time: Option<SystemTime>,
-    // subscribers for when the task is aborted, succeeded, etc.
-    subscribers: Mutex<Vec<OneshotSender<TaskWaiterResult>>>,
+    // TODO: find a way for this not to be in a mutex.
```

Comment:
👍 scrolled back here to say this

Reply:
Yeah, I think this has more to do with task spawning patterns than anything else. Any thread is creating tasks rather than there being strict single thread doing it, so it ends up needing Send/Sync.  So, yeah, the mutex will go, but will require a separate scheduler thread to process the submission requests.